utils/appliance/__init__.py

```diff
@@ -1931,64 +1927,48 @@ def host_id(self, hostname):
 
     @cached_property
     def is_storage_enabled(self):
-        return 'storage' in self.get_yaml_config('vmdb').get('product', {})
+        return 'storage' in self.get_yaml_config().get('product', {})
 
-    def get_yaml_config(self, config_name):
-        if config_name == 'vmdb':
-            writeout = self.ssh_client.run_rails_command(
-                '"File.open(\'/tmp/yam_dump.yaml\', \'w\') '
-                '{|f| f.write(Settings.to_hash.deep_stringify_keys.to_yaml) }"'
-            )
-            if writeout.rc:
-                logger.error("Config couldn't be found")
-                logger.error(writeout.output)
-                raise Exception('Error obtaining config')
-            base_data = self.ssh_client.run_command('cat /tmp/yam_dump.yaml')
-            if base_data.rc:
-                logger.error("Config couldn't be found")
-                logger.error(base_data.output)
-                raise Exception('Error obtaining config')
-            try:
-                return yaml.load(base_data.output)
-            except:
-                logger.debug(base_data.output)
-                raise
-        else:
-            raise Exception('Only [vmdb] config is allowed from 5.6+')
-
-    def set_yaml_config(self, config_name, data_dict):
-        if config_name == 'vmdb':
-            temp_yaml = NamedTemporaryFile()
-            dest_yaml = '/tmp/conf.yaml'
-            yaml.dump(data_dict, temp_yaml, default_flow_style=False)
-            self.ssh_client.put_file(temp_yaml.name, dest_yaml)
-            # Build and send ruby script
-            dest_ruby = '/tmp/set_conf.rb'
-
-            ruby_template = data_path.join('utils', 'cfmedb_set_config.rbt')
-            ruby_replacements = {
-                'config_file': dest_yaml
-            }
-            temp_ruby = load_data_file(ruby_template.strpath, ruby_replacements)
-            self.ssh_client.put_file(temp_ruby.name, dest_ruby)
+    def get_yaml_config(self):
+        writeout = self.ssh_client.run_rails_command(
+            '"File.open(\'/tmp/yam_dump.yaml\', \'w\') '
+            '{|f| f.write(Settings.to_hash.deep_stringify_keys.to_yaml) }"'
+        )
+        if writeout.rc:
+            logger.error("Config couldn't be found")
+            logger.error(writeout.output)
+            raise Exception('Error obtaining config')
+        base_data = self.ssh_client.run_command('cat /tmp/yam_dump.yaml')
```

Comment:
OPTIONAL: we have the number of commands which we run over ssh everywhere. I'd create an SSHAgent with pre-defined commands